department-of-veterans-affairs/va.gov-cms
# DNM: SPIKE: VAGOV-TEAM-83275 Webform Viability

## Description

Relates to [#83275](https://github.com/department-of-veterans-affairs/va.gov-team/issues/83275)

investigating the webform suite of modules for viability for the Form Engine team's proposed solution

## Testing done
see notes in ticket for more detail.

looking into form patterns, custom elements, outputting form data, permissions, moderation workflows, etc


## Screenshots
an example form ui, nested with a custom chapter + page elements
![Screenshot 2024-05-29 at 2 04 26 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/11279744/71a2c27d-607c-4a6d-8451-9398de83fc65)

category for custom elements
![Screenshot 2024-05-29 at 1 59 40 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/11279744/3376b6a1-7ca3-4c8b-9d08-d44c1bcfc982)

what the composite example looks like in the drupal ui
![Screenshot 2024-05-29 at 2 04 11 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/11279744/1bd966d5-5f32-4ed1-b7d0-ea50d2cf2a7f)


## QA steps

go to `/admin/structure/webform` and create a new webform. name it whatever you want. 

add elements to your form. **you need to use the VA Chapter and VA Page if you want to nest wizard pages. the default "Add Page" element does not do this!**

once you have elements added to a form in a manner you're satisfied with, verify that the jsonschema for the form is also generated at:
`/webform_jsonschema/your-form-name?_format=webform_jsonschema`


### Select Team for PR review


### Is this PR blocked by another PR?
